File: 29-04-2026/url_query_parser.js
function parseUrlQuery(url) {
  let [query_string, result] = [url.split("?")[1].split("&") || [], {}];

  for (let pair of query_string) {
    let [key, value] = pair.split("=");
    result[key] = value;
  }

  return result;
}

console.log(parseUrlQuery("https://example.com/search?name=Alice&age=30")); // {"name": "Alice", "age": "30"}
console.log(
  parseUrlQuery(
    "https://freecodecamp.org/learn?skill=programming&language=python",
  ),
); // {"skill": "programming", "language": "python"}
console.log(
  parseUrlQuery(
    "https://freecodecamp.org/items?category=books&sort=asc&page=2",
  ),
); // {"category": "books", "sort": "asc", "page": "2"}
console.log(
  parseUrlQuery("https://example.com?redirect=freecodecamp.org/learn&when=now"),
); // {"redirect": "freecodecamp.org/learn", "when": "now"}

File: 29-04-2026/url_query_parser.py
def parse_url_query(url):
    query_string = url.split('?')[1].split('&') if '?' in url else ''
    result = {}

    for pair in query_string:
        key, value = pair.split('=')
        result[key] = value

    return result

print(parse_url_query("https://example.com/search?name=Alice&age=30")) # {"name": "Alice", "age": "30"}
print(parse_url_query("https://freecodecamp.org/learn?skill=programming&language=python")) # {"skill": "programming", "language": "python"}
print(parse_url_query("https://freecodecamp.org/items?category=books&sort=asc&page=2")) # {"category": "books", "sort": "asc", "page": "2"}
print(parse_url_query("https://example.com?redirect=freecodecamp.org/learn&when=now")) # {"redirect": "freecodecamp.org/learn", "when": "now"}